### ISSUE TITLE
"git [pull | someNonsense]" hang on Windows 10 when I run from git bash (from cmd.exe it's fine)

### ISSUE THREAD

**Issue by lll000111** [git-for-windows/git#1634]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

I found an issue on SO that sounds similar ("git pull" hanging after it's already done its job was my original problem) but has no solution: https://stackoverflow.com/questions/44031150/why-does-git-pull-hang

### Setup

 - Which version of Git for Windows are you using? 

git version 2.17.0.windows.1
```

$ git --version --build-options

git version 2.17.0.windows.1
cpu: x86_64
built from commit: e7621d891d081acff6acd1f0ba6ae0adce06dd09
sizeof-long: 4
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Windows 10 64bit

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.16299.371]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
$ cat /etc/install-options.txt

Editor Option: Notepad++
Path Option: BashOnly
SSH Option: OpenSSH
CURL Option: WinSSL
CRLF Option: LFOnly
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

None that I'm aware of.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

git bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

Run something nonsensical from git bash:
```
git uuuuuuuuuuu
```

**But** I originally had that issue with "`git pull`" - it does the complete pull (the repo status is clean and with the pulled files afterwards) but the command never returns.

This has only been happening for a few days.

**CTRL-C has no effect**

 - What did you expect to occur after running these commands?

I expected 
> git "uuuuuuuuuuu" is not a git command

 - What actually happened instead?

It hangs forever (I left it up and saw no timeout even after half an hour):

```
$ GIT_TRACE=true git uuuuuuuuuuuu
14:33:12.478891 git.c:576               trace: exec: git-uuuuuuuuuuuu
14:33:12.480894 run-command.c:640       trace: run_command: git-uuuuuuuuuuuu
[HANGS...]
```

When I run `"C:\Program Files\Git\bin\git.exe" somenonsense` in a Windows command shell I immediately get the expected error message.

The problem occurred on two different Windows 10 systems, both my laptop and my PC, which are independent (and "just accept defaults") setups . I don't run any tools, it's mostly just git/bash, ConEmu, Chrome, Firefox, WebStorm and not much else. I like to keep my systems basic and bare. Onedrive is enabled but the issue occurs when I run git anywhere, just using the nonsense command. I think I actually had git 16.2 still installed on the laptop when I first tried, since it did not update itself to 17 for some unknown reason (unlike the desktop git) I manually installed git 17 only after I first tried the above on the older version.

Since this has been happening only for a few days (at some point this week) I guess it could have been one of the recent Windows patches rather than a change in git or bash?

**Or** was it one of the recent ConEmu updates? But how...?

**Comment by lll000111** [comment]
UPDATE: I think it's one of the recent ConEmu updates (this week, two at least). When I run bash.exe on top of a Windows cmd.exe console and then run the command that hangs it works as expected.

**Comment by lll000111** [comment]
Related? https://github.com/Maximus5/ConEmu/issues/1510
